File: Java/Easy/Accepted/duplicate-zeros_4.java
class Solution {
    public void duplicateZeros(int[] arr) {
        
        int n = arr.length;
        int[] newArr = new int[n];
        int i = 0;
        int j = 0;
        while (i < n) {
            if (arr[j] == 0) {
                newArr[i++] = 0;
                if (i < n) {
                    newArr[i++] = 0;
                }
            } else {
                newArr[i++] = arr[j];
            }
            j++;
        }
        for (i = 0; i < n; i++) {
            arr[i] = newArr[i];
        }
    }
}

File: C++/Easy/Accepted/duplicate-zeros_4.cpp
class Solution {
public:
    void duplicateZeros(vector<int>& arr) {
        
        int len = arr.size();
        int i = 0;
        while (i < len) {
            if (arr[i] == 0) {
                for (int j = len - 1; j > i; --j) {
                    arr[j] = arr[j-1];
                }
                i += 2;
            } else {
                i++;
            }
        }
    }
};